docs/source/new-docs/tutorials/2024-tutorial.rst

```diff
@@ -0,0 +1,515 @@
+
+#. Docstring
+
+   A ``docstring`` is a chunk of documentation which describes what your plugin does. This
+   property is required for every GeoIPS plugin created, module-based or YAML-based. We
+   require this property for proper documentation of created plugins, and it will also be
+   a useful feature later on when the GeoIPS Command Line Interface (CLI) is created, as
+   you will be able to see what each plugin does provided the ``docstring`` for that plugin
+   is filled.
+
+.. _plugin-development-setup:
+
+Product Plugin Development Initial Setup
+****************************************
+
+Before creating a new product for CLAVR-x Cloud-Top-Height, let's get the initial setup done:
+
+#. To develop a new GeoIPS plugin, install :ref:`GeoIPS<linux-installation>` and make sure that
+   you have ``geoips`` Python environment enabled throughout this tutorial using
+
+   .. code-block:: shell
+
+    mamba activate geoips # activating python envrionment
+
+   You will know if geoips environment is enabled if it shows up ahead of your username in your command prompt.
+
+#. Next, let's install GeoIPS CLAVR-x package and test the installation. This is needed as we are developing products
+for GeoIPS CLAVR-x.
+
+   .. code-block:: shell
+
+    git clone https://github.com/NRLMMD-GEOIPS/geoips_clavrx $GEOIPS_PACKAGES_DIR/geoips_clavrx # download the remote
+    repository
+    pip install -e $GEOIPS_PACKAGES_DIR/geoips_clavrx # installing the geoips_clavrx
+
+    $GEOIPS_PACKAGES_DIR/geoips/setup/check_system_requirements.sh test_data test_data_clavrx      # Install the clavrx
+    test data repo
+    $GEOIPS_PACKAGES_DIR/geoips_clavrx/tests/test_all.sh  # Run tests to verify geoips-clavrx installation
+
+#. Now, set the following additonal environment variables which are specific to your product plugin development
+
+   .. code-block:: shell
+
+      export MY_PKG_NAME=<your package name>    #read the note below for your package name,
+      export MY_PKG_DIR=$GEOIPS_PACKAGES_DIR/$MY_PKG_NAME    #your package directory
+      export MY_PKG_URL=<your package’s URL on version control platform(GitLab)> #your package VCS url
+
+   .. NOTE::
+    Choose a name for your package making sure that it is in lower case, starting with a letter,
+    and only contains letters, numbers, and underscores.
+
+#. Navigate to your product plugin directory and clone the example repository of customized plugin development,
+`Template Basic Plugin <https://github.com/NRLMMD-GEOIPS/template_basic_plugin/tree/main>`_ that would guide us through
+the process of creating a new plugin package containing one or more custom plugins.
+
+   .. code-block:: shell
+
+      cd $GEOIPS_PACKAGES_DIR         #Go to your package directory
+      git clone --no-tags --single-branch $GEOIPS_REPO_URL/template_basic_plugin.git
+
+   .. NOTE::
+    If you're not able to move into the directory listed in the above code-block. Verify if the values of
```

Comment:
"If you're not able to move into the directory listed in the above code-block." This is a dependent clause and cannot stand alone as a sentence. It needs to be connected to the following part to form a complete sentence.

There is a period after "code-block" that creates a sentence fragment.

The sentence lacks proper punctuation to separate different parts and ideas, which makes it hard to read and understand.

Reply:
The phrase "the values of environment variable(s) is/are" is awkward. It’s trying to account for both singular and plural, but it ends up being confusing. Choose one form or rewrite it for clarity.

Reply:
The part "otherwise check the step three again and if needed take help" lacks parallel structure and clear punctuation. It should be clearer and more structured.

Reply:
Here is a revised version of the sentence with these issues corrected:

"If you are not able to move into the directory listed in the above code block, verify if the values of the environment variables are set using the command shown below. Otherwise, check step three again and seek help if needed."